api/src/main/java/eu/darkbot/api/managers/QuestAPI.java

```diff
@@ -0,0 +1,194 @@
+package eu.darkbot.api.managers;
+
+import eu.darkbot.api.API;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+/**
+ * Provides access to data for the Quests in-game.
+ */
+public interface QuestAPI extends API.Singleton {
+
+    /**
+     * @return The quest that is currently active and in view.
+     *         Even if there are several accepted quests, only the selected one is
+     *         returned.
+     */
+    @Nullable Quest getCurrentQuestDisplayed();
+
+    /**
+     * @return The quest that is currently selected in the QuestGiver
+     */
+    @Nullable Quest getCurrentQuestSeleted();
+
+    /**
+     * @return Returns additional information on the selected quest
+     */
+    @Nullable QuestListItem getCurrentQuestInfoSeleted();
```

Comment:
```suggestion
    @Nullable QuestListItem getSelectedQuestInfo();
```
Question, shouldn't QuestListItem be a "less info" version of Quest? if so, can one extend the other?
and if they do extend, then the previous method would have all the info of this one, so i'd make it obsolete.

Maybe consider at least a minimum BaseQuest interface with whatever is common, then Quest and QuestListItem for the two more specialized interfaces with extra info

Reply:
QuestListItem is the item that is shown in the quest giver list and according to what I have been seeing the only thing they share is the id but they do not have the same offsets. That is why I believe it should be treated differently